### PR TITLE
Examples: Added MDDLoader.

### PR DESCRIPTION
**Pull request by Mugen87**
Added a loader for the `MDD` format which can be used to exchange animation data between 3D authoring apps and engines. It's not very common but I've noticed a few users in the past who looked for the possibility to directly load the data into their app.

The `MDD` format is very simple and thus the loader's implementation.

### CODE REVIEW

**Comment by mrdoob** [comment]
Nice! Did you come up with this format? Or is there a design/spec somewhere?

**Comment by Mugen87** [comment]
> Or is there a design/spec somewhere?

It's a proprietary format and available in different tools like Modo or Blender. 

https://learn.foundry.com/modo/content/help/pages/system_files/export_mdd.html

Unfortunately, there is no official specification. It took a while to figure out how a MDD file is structured. Fortunately, [other communities](https://community.foundry.com/discuss/topic/37912/pc2-to-mdd-max-to-modo-beta-available?mode=Post&postID=609598) already discussed about this topic.

**Comment by mrdoob** [comment]
Oh! Doh!

I now remember that I was meaning to look into this format because that's the original format I have the ROME models in 😅

**Comment by mrdoob** [comment]
Great stuff!

**Comment by mrdoob** [comment]
Thanks!

File: examples/webgl_loader_mdd.html [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+				clock = new THREE.Clock();
+
+				//
+
+				var hemiLight = new THREE.HemisphereLight( 0xffffff, 0x444444 );
+				hemiLight.position.set( 0, 20, 0 );
+				scene.add( hemiLight );
+
+				var dirLight = new THREE.DirectionalLight( 0xffffff );
+				dirLight.position.set( - 3, 10, - 10 );
```
Comment (WestLangley) [inline examples/webgl_loader_mdd.html]: This light is shining from the back.

Depending on how you choose to fix this, try to avoid an over-bright scene.

Comment (mrdoob) [reply examples/webgl_loader_mdd.html]: I thought it was intentional?

Comment (WestLangley) [reply examples/webgl_loader_mdd.html]: It adds no light to the scene.

Comment (Mugen87) [reply examples/webgl_loader_mdd.html]: Well, the top side is a little bit brighter with the directional light.

TBH, I think it's better to remove all lights and use `MeshNormalMaterial` instead.

Comment (Mugen87) [reply examples/webgl_loader_mdd.html]: 42c09065d951c526702b545fba016ce5e2a7f1e9

File: examples/webgl_loader_mdd.html [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+				//
+
+				var loader = new MDDLoader();
+				loader.load( 'models/mdd/cube.mdd', function ( result ) {
+
+					const morphTargets = result.morphTargets;
+					const clip = result.clip;
+					// clip.optimize(); // optional
+
+					var geometry = new THREE.BoxBufferGeometry();
```
Comment (mrdoob) [inline examples/webgl_loader_mdd.html]: Do I understand it correctly that a MDD file has no information of how points connect to each other?

Comment (Mugen87) [reply examples/webgl_loader_mdd.html]: Yes, the MDD files has to exactly match the used geometry. Meaning even the order of vertices must be equal.

Comment (mrdoob) [reply examples/webgl_loader_mdd.html]: Got it. Thanks!

**Pull request merged**